File: intro/spring/security/framework/src/main/java/org/tsdes/intro/spring/security/framework/db/UserEntity.java
package org.tsdes.intro.spring.security.framework.db;

import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.Set;

/**
 * Created by arcuri82 on 13-Dec-17.
 */
@Entity
@Table(name="USERS")
public class UserEntity {

    @Id
    @NotBlank
    private String username;

    @NotBlank
    private String password;

    @ElementCollection
    private Set<String> roles;

    @NotNull
    private Boolean enabled;

    public UserEntity() {
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Set<String> getRoles() {
        return roles;
    }

    public void setRoles(Set<String> roles) {
        this.roles = roles;
    }

    public Boolean getEnabled() {
        return enabled;
    }

    public void setEnabled(Boolean enabled) {
        this.enabled = enabled;
    }
}

File: spring/security/database/src/main/kotlin/org/tsdes/spring/security/database/db/UserEntity.kt
package org.tsdes.spring.security.database.db

import org.hibernate.validator.constraints.NotBlank
import javax.persistence.ElementCollection
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table
import javax.validation.constraints.NotNull

/**
 * Created by arcuri82 on 08-Nov-17.
 */
@Entity
@Table(name="USERS")
class UserEntity(

        @get:Id
        @get:NotBlank
        var username: String?,

        @get:NotBlank
        var password: String?,

        @get:ElementCollection
        @get:NotNull
        var roles: Set<String>? = setOf(),

        @get:NotNull
        var enabled: Boolean? = true
)